src/Elements/Input.react.js

```diff
@@ -207,15 +207,21 @@ class Input extends Component {
                         newValue = Math.max(min, newValue);
                     }
                 }
-            }
+
+                this.setState({ value: newValue });
+
+                if (!_.isUndefined(onChange)) {
+                    onChange(newValue);
+                }
+            }, 500);
+            this.setState({ value: newValue });
+        } else {
+            this.setState({ value: newValue });
```

Comment:
this.setState({ value: newValue }); needs to be in an else if onChange` is undefined.

Reply:
Fixed